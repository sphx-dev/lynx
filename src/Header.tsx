import Container from "react-bootstrap/Container"
import Navbar from "react-bootstrap/Navbar"
import ReadMe from "./ReadMe"
import logo from "./static/logo.png"

function Header() {
  return (
    <>
      <Navbar bg="dark" data-bs-theme="dark">
        <Container fluid={true}>
          <Navbar.Brand href="#home">
            <img src={logo} height="60" alt="header-logo" />
          </Navbar.Brand>
          <Navbar.Collapse className="justify-content-end">
            <ReadMe />
          </Navbar.Collapse>
        </Container>
      </Navbar>
    </>
  )
}

export default Header
