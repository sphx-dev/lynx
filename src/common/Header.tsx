import { FC } from "react"
import styled from "styled-components"
import { ConnectButton } from "@rainbow-me/rainbowkit"
import { FlexDivCol } from "../components/Flex"
import logo from "../assets/logo.png"
// import WalletButtons from "./WalletButtons"

const Header: FC = () => {
  return (
    <FlexDivCol>
      <Container>
        <LogoNav>
          <img src={logo} height="60" alt="header-logo" />
        </LogoNav>
        <ConnectButton />
      </Container>
    </FlexDivCol>
  )
}

const Container = styled.header`
  display: flex;
  justify-content: space-between;
  padding: 15px;
`

const LogoNav = styled.div`
  display: flex;
  align-items: center;
`

export default Header
