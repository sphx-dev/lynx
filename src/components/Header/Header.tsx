import styled from "styled-components";
import logout from "../../assets/icons/logout.svg";
import myAccount from "../../assets/icons/account.svg";
import logo from "../../assets/icons/logo-dark.svg";
import menu from "../../assets/icons/menu.svg";
import Group from "../Group";
import Nav from "../../sections/navbar";

const StyledHeader = styled.header`
  position: relative;
  background-image: url("/src/assets/images/header-bg.png");
  background-position: top;
  width: 320px;
  height: 100px;
  margin-top: -5px;
`;

const NavBar = styled.nav`
  position: absolute;
  top: 70px;
  display: flex;
  justify-content: space-evenly;
  width: 80%;
`;
const Button = styled.div`
  color: ${({ theme }) => theme.colors.common.palette.secondary["700"]};
  ${({ theme }) => theme.fonts.typography.header};
  cursor: pointer;
  display: flex;
  img {
    margin-right: 4px;
    margin-bottom: 2px;
    height: 14px;
  }
  &:hover {
    color: ${({ theme }) => theme.colors.common.palette.secondary["100"]};
  }
`;

const Logo = styled.img`
  padding-top: 12px;
  margin-left: 64px;
`;

const Menu = styled.img`
  padding-top: 16px;
  margin-left: 16px;
`;

const Divider = styled.div`
  border-left: 1px solid #173d46;
`;
const Header = () => {
  return (
    <Group fluid spacing={0} style={{ width: "100vw" }}>
      <StyledHeader>
        {/*<Menu src={menu} />*/}
        <Logo src={logo} />
        <NavBar>
          <Button>
            <img src={myAccount} alt="my account" />
            MY ACCOUNT
          </Button>
          <Divider />
          <Button>
            <img src={logout} alt="my account" />
            LOGOUT
          </Button>
        </NavBar>
        {/*<span>Logout</span>*/}
      </StyledHeader>
      <Nav />
    </Group>
  );
};
export default Header;
