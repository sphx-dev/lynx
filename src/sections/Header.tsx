import Container from "react-bootstrap/Container"
import Navbar from "react-bootstrap/Navbar"
import Dropdown from "react-bootstrap/Dropdown"
import Badge from "react-bootstrap/Badge"
import ReadMe from "../ReadMe"
import { FaBell } from "react-icons/fa"
import logo from "../static/logo.png"

function Header() {
  return (
    <>
      <Navbar bg="dark" data-bs-theme="dark">
        <Container fluid={true}>
          <Navbar.Brand href="#home">
            <img src={logo} height="60" alt="header-logo" />
          </Navbar.Brand>
          <Navbar.Collapse className="justify-content-end">
            <Dropdown>
              <Dropdown.Toggle variant="primary" id="dropdown-basic">
                Connected 0x21837adc9...
                <Badge bg="danger" style={{ marginLeft: "1rem" }}>
                  <FaBell /> 2
                </Badge>
              </Dropdown.Toggle>
              <Dropdown.Menu>
                <Dropdown.Item href="#/action-1">Wallet</Dropdown.Item>
                <Dropdown.Item href="#/action-1">Reports</Dropdown.Item>
                <Dropdown.Item href="#/action-2">Settings</Dropdown.Item>
                <Dropdown.Item href="#/action-3">Logout</Dropdown.Item>
              </Dropdown.Menu>
            </Dropdown>
            <ReadMe />
          </Navbar.Collapse>
        </Container>
      </Navbar>
    </>
  )
}

export default Header
