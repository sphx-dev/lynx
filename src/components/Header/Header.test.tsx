import { render } from "@testing-library/react";
import Header from "./Header";
import "@testing-library/jest-dom";
import { AppWrapper } from "@/testing/AppWrapper";

describe("Header Component", () => {
  it("should render the Header component correctly", () => {
    const { container } = render(<Header />, {
      wrapper: AppWrapper,
    });

    expect(container).toMatchSnapshot();
  });

  it("should render the Nav component", () => {
    const { getByRole } = render(<Header />, { wrapper: AppWrapper });

    // Check if the Nav component is rendered
    const nav = getByRole("navigation");
    expect(nav).toBeInTheDocument();

    // Snapshot test for the Nav component
    expect(nav).toMatchSnapshot();
  });
});
