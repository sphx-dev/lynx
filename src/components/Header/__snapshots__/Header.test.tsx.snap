// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Header Component > should render the Header component correctly 1`] = `
<div>
  <div>
    <button
      data-testid="connect-test-button"
    />
  </div>
  <div
    align="start"
    class="sc-blHHSb jWUilC"
    position="left"
    spacing="0"
    style="width: 100vw;"
  >
    <header
      class="sc-dSfWjt kRGbXl"
    >
      <img
        class="sc-gohMHu icTzcf"
        src="/src/assets/icons/logo-dark.svg"
      />
    </header>
    <nav
      class="sc-hSyjfr gEUtBX"
    >
      <a
        class="sc-kLJHhQ ansjc"
        href="/dashboard"
      >
        Dashboard
      </a>
      <a
        class="sc-kLJHhQ ansjc"
        href="/exchange"
      >
        Exchange
      </a>
      <a
        class="sc-kLJHhQ ansjc"
        href="/dashboard/faucet"
      >
        Faucet
      </a>
      <a
        class="sc-kLJHhQ ansjc"
        href="https://station.non-prod.sphx.dev"
        target="_blank"
      >
        Station
      </a>
      <a
        class="sc-kLJHhQ ansjc"
        href="https://explorer.sphx.dev"
        target="_blank"
      >
        Explorer
      </a>
      <button
        class="sc-fAUdSK iHBsqq"
        variant="secondary"
      >
        Add Chain
      </button>
      <span
        class="sc-dntaoT hkozYz"
      >
        <button
          class="sc-fAUdSK gxMJIT"
          variant="primary"
        >
          Connect
        </button>
      </span>
    </nav>
  </div>
  <div
    id="modal-root"
  />
</div>
`;

exports[`Header Component > should render the Nav component 1`] = `
<nav
  class="sc-hSyjfr gEUtBX"
>
  <a
    class="sc-kLJHhQ ansjc"
    href="/dashboard"
  >
    Dashboard
  </a>
  <a
    class="sc-kLJHhQ ansjc"
    href="/exchange"
  >
    Exchange
  </a>
  <a
    class="sc-kLJHhQ ansjc"
    href="/dashboard/faucet"
  >
    Faucet
  </a>
  <a
    class="sc-kLJHhQ ansjc"
    href="https://station.non-prod.sphx.dev"
    target="_blank"
  >
    Station
  </a>
  <a
    class="sc-kLJHhQ ansjc"
    href="https://explorer.sphx.dev"
    target="_blank"
  >
    Explorer
  </a>
  <button
    class="sc-fAUdSK iHBsqq"
    variant="secondary"
  >
    Add Chain
  </button>
  <span
    class="sc-dntaoT hkozYz"
  >
    <button
      class="sc-fAUdSK gxMJIT"
      variant="primary"
    >
      Connect
    </button>
  </span>
</nav>
`;
